src/proto/nc_redis.c

```diff
@@ -2912,10 +2926,23 @@ redis_post_connect(struct context *ctx, struct conn *conn, struct server *server
 
     /* enqueue as head and send */
     req_server_enqueue_imsgq_head(ctx, conn, msg);
-    msg_send(ctx, conn);
+    *sended = 1;
 
     log_debug(LOG_NOTICE, "sent 'SELECT %d' to %s | %s", pool->redis_db,
               pool->name.data, server->name.data);
+
+    return NC_OK;
+}
+
+void
+redis_post_connect(struct context *ctx, struct conn *conn, struct server *server)
+{
+    int sended = 0;
+    redis_select_db(ctx, conn, server, &sended);
```

Comment:
```
    /* enqueue as head and send */
    req_server_enqueue_imsgq_head(ctx, conn, msg);
```

Oh, I see, both of these will set the new message to be the head of the queue. So I misread it at first, it's actually sending the command to add auth before selecting the db number.

(though I assume moving the auth check out of req_forward was the actual fix)

Reply:
> it's actually sending the command to add auth before selecting the db number.

Exactly